Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -509,10 +520,12 @@ inline void SeedFinder<external_spacepoint_t, platform_t>::filterCandidates(
     float sinTheta = 1 / std::sqrt(iSinTheta2);
     float cosTheta = cotThetaB * sinTheta;
 
-    // clear all vectors used in each inner for loop
-    state.topSpVec.clear();
```

Comment:
`clear` does not in fact reduce the capacity, so this ~ only resets the last-element index to 0.